androidplot-core/src/main/java/com/androidplot/xy/PanZoom.java

```diff
@@ -347,7 +378,14 @@ protected void calculateZoom(RectF newRect, float scale, boolean isHorizontal) {
                 }
             }
         } else {
-            final RectRegion limits = plot.getOuterLimits();
+            // zoom limited and increment by value StepMode?
+            if (zoomLimit == ZoomLimit.MIN_TICKS && (plot.getRangeStepMode() == StepMode.INCREMENT_BY_FIT || plot.getRangeStepMode() == StepMode.INCREMENT_BY_VAL)) {
```

Comment:
I have not had a chance to look super closely yet, but I'm wondering if there's a way to avoid inspecting the StepMode state to accomplish this effect.